Prox/Prox/Utilities/CategoriesUtil.swift

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+struct CategoriesUtil {
+
+    static let HiddenRootCategories = Set(arrayLiteral: "auto", // 🚗
+                                                  "bicycles",
+                                                  "education",
+                                                  "financialservices",
+                                                  "health",
+                                                  "homeservices",
+                                                  "localservices",
+                                                  "professional",
+                                                  "realestate",
+                                                  "religiousorgs")
+
+    private static let AllCategoriesPath = "Data.bundle/yelp_categories_v3"
+    private static let AllCategoriesExt = "json"
+
+    static func shouldShowPlace<S : Sequence>(byCategories categories: S) throws -> Bool where S.Iterator.Element == String {
+        // Hide if all match.
+        let rootCategories = try getRootCategories(forCategories: categories)
+        let allCategoriesMatch = rootCategories.subtracting(HiddenRootCategories).isEmpty
+        return !allCategoriesMatch
+    }
+
+    // This accesses the app bundle, which could be slow. If it's an issue, consider using a background thread.
+    static let categoryToParents = getCategoryToParents()
+
+    private static func getCategoryToParents() -> [String:Set<String>] {
```

Comment:
Changed public var to `categoriesToParentsMap` and the function to `getCategoriesToParentsMap`.